feat(curriculum) : Add Asynchronous JavaScript Quiz (#56685)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-asynchronous-javascript/66edd630f7666cfa54b404d0.md

#### --text--

What is asynchronous programming?

#### --distractors--

A way to execute code in a sequential order.

---

A method to execute code concurrently without blocking the main thread.

---

A technique to handle errors in synchronous code.

#### --answer--

A method to execute code concurrently without blocking the main thread.

### --question--

#### --text--

What does the `async` attribute do when used in a `<script>` tag?

#### --distractors--

It allows the script to be executed after the HTML document has been completely parsed.

---

It makes the script execute immediately, regardless of the order of script tags

---

It ensures the script runs in a synchronous manner.

#### --answer--

> It allows the script to be executed after the HTML document has been completely parsed.

Reviewer: Incorrect, that is `defer`

#### --distractors--

`GET`

---

`POST`

---

`DELETE`

#### --answer--

`POST`

### --question--

#### --text--

What does the `res.json()` method do in Express.js?

#### --distractors--

It converts a JavaScript object to a JSON string and sends it as a response.

---

It retrieves data from a JSON file.

---

It parses a JSON string into a JavaScript object.

#### --answer--

It converts a JavaScript object to a JSON string and sends it as a response.

### --question--

#### --text--

What is a promise in JavaScript?

#### --distractors--

A way to synchronize multiple operations.

---

A callback function that runs after an operation.

---

An object that represents the eventual completion (or failure) of an asynchronous operation.

#### --answer--

An object that represents the eventual completion (or failure) of an asynchronous operation.

### --question--

#### --text--

What does promise chaining allow you to do?

#### --distractors--

Execute multiple asynchronous operations in parallel.

---

Handle synchronous operations more efficiently.

---

Pass the result of one promise to the next.

#### --answer--

> Pass the result of one promise to the next.

Reviewer: these two are equal, all four options should be different from each other

### --question--

#### --text--

Which method is used to handle successful completion of a promise?

#### --distractors--

`.finally()`

---

`.catch()`

---

`.error()`

#### --answer--

`.then()`

### --question--

#### --text--

Which method is used to handle errors in a promise chain?

#### --distractors--

`.finally()`

---

`.then()`

---

`.error()`

#### --answer--

`.catch()`

### --question--

#### --text--

> What is the use of `.finally()` method in a promise?

Reviewer 2: ```suggestion
What is the use of the `.finally()` method in a promise?
```

It returns the result of the promise.

#### --answer--

It pauses the execution of the function until the promise is resolved.

### --question--

#### --text--

What is the purpose of the Geolocation API?

#### --distractors--

To manipulate HTML elements.

---

To create and manage map elements.

---

To control the device's camera.

#### --answer--

To retrieve the geographical location of a device.

### --question--

#### --text--

Which method of the Geolocation API is used to get the current position of the device?

#### --distractors--

`getLocation()`

---

`getPosition()`

---

`getLatLong()`

#### --answer--

`getCurrentPosition()`

### --question--

#### --text--

What parameters can be passed to the `getCurrentPosition()` method?

#### --distractors--

Only the success callback function.

---

Only options.

---

One the error callback function.

#### --answer--

Success and error callback functions, along with options.

### --question--

#### --text--

What will the following code return? `fetch('https://api.example.com/data')`

#### --distractors--

The JSON response directly.

---

An error if the URL is not reachable.

---

A synchronous response.

#### --answer--

> A Promise that resolves to Response object

Reviewer 2: ```suggestion
A Promise that resolves to a Response object
```

> A Promise that resolves to a Response object

Reviewer 3: ```suggestion
A `Promise` that resolves to a `Response` object
```
Since these refer in example to a concrete promise and response objects, maybe they should be backticked...

#### --text--

What is asynchronous programming?

#### --distractors--

A way to execute code in a sequential order.

---

A method to execute code concurrently without blocking the main thread.

---

A technique to handle errors in synchronous code.

---

> A way to guarantee task order.

Reviewer: do not add an extra distractor, replace the one that is equal to answer, you should have 4 choices total, 3 distractors and one correct answer

#### --answer--

A method to execute code concurrently without blocking the main thread.

### --question--

#### --text--

What does the `async` attribute do when used in a `<script>` tag?

#### --distractors--

It allows the script to be executed after the HTML document has been completely parsed.

---

It makes the script execute immediately, regardless of the order of script tags.

---

It ensures the script runs in a synchronous manner.

---

It delays the script execution until the browser is idle.

#### --answer--

> It makes the script execute immediately, regardless of the order of script tags.

Reviewer: ```suggestion
It makes the script execute as soon as it is loaded, regardless of the order of script tags.
```

Reviewer: This is still an issue. Your answer is describing `defer`, not `async`
